libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraComponent.kt

```diff
@@ -1,147 +1,186 @@
 package com.mapbox.navigation.dropin.component.camera
 
-import android.location.Location
-import com.mapbox.android.gestures.MoveGestureDetector
+import com.mapbox.maps.EdgeInsets
 import com.mapbox.maps.MapView
 import com.mapbox.maps.plugin.animation.camera
-import com.mapbox.maps.plugin.gestures.OnMoveListener
-import com.mapbox.maps.plugin.gestures.gestures
 import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.DropInNavigationViewContext
 import com.mapbox.navigation.dropin.component.location.LocationViewModel
+import com.mapbox.navigation.dropin.component.navigationstate.NavigationState
 import com.mapbox.navigation.dropin.extensions.flowNavigationCameraState
 import com.mapbox.navigation.dropin.extensions.flowRouteProgress
 import com.mapbox.navigation.dropin.extensions.flowRoutesUpdated
 import com.mapbox.navigation.dropin.lifecycle.UIComponent
 import com.mapbox.navigation.ui.maps.camera.NavigationCamera
 import com.mapbox.navigation.ui.maps.camera.data.MapboxNavigationViewportDataSource
+import com.mapbox.navigation.ui.maps.camera.lifecycle.NavigationBasicGesturesHandler
 import com.mapbox.navigation.ui.maps.camera.state.NavigationCameraState
 import kotlinx.coroutines.flow.collect
-import kotlinx.coroutines.flow.filterNotNull
 import kotlinx.coroutines.launch
 
 internal class CameraComponent constructor(
     private val mapView: MapView,
+    private val navigationCamera: NavigationCamera,
+    private val viewportDataSource: MapboxNavigationViewportDataSource,
+    private val dropInNavigationViewContext: DropInNavigationViewContext,
```

Comment:
Assuming this is temporary? We will use NavigatioStateViewModel after this merges?
https://github.com/mapbox/mapbox-navigation-android/pull/5506

Reply:
Yes